### PR TITLE
Consultation report

### PR DESCRIPTION
**Pull request by martgnz**
Connects to #359.

### What does this PR do?
This PR implements the new budget consultation report on the admin module.

It features statistics about the current consultations, a bar chart with participant's ages, and a table summarizing the response to each budget line proposal.

### Figures
- Total responses
- Gender breakdown
- Budget outcome summary
- Average age
- Location ranking
- Age report
- Budget line responses

### CODE REVIEW

File: app/assets/javascripts/admin/application.js [hunk 1]
```diff
@@ -16,6 +16,16 @@
 
 $(document).on('turbolinks:load', function() {
 
+  if ($('#age_report').length) {
```
Comment (furilo) [inline app/assets/javascripts/admin/application.js]: We already have a admin/gobierto_budget_consultations/application.js - lets put this there.

File: app/assets/stylesheets/module-admin.scss [hunk 2]
```diff
@@ -412,3 +412,141 @@ trix-toolbar .button_group button.icon, trix-toolbar .button_group input[type=bu
 	width: 2em;
 	font-size: 14px;
 }
+
+
+/* Consultations report */
+div[data-tab="consultation_report"] {
```
Comment (furilo) [inline app/assets/stylesheets/module-admin.scss]: Tabs instead of spaces?

Comment (furilo) [inline app/assets/stylesheets/module-admin.scss]: Look at the working integration, and you'll see that navigation is not with tabs but over full page reload, so this won't work. Just use a class name for the container.

File: app/assets/stylesheets/module-admin.scss [hunk 3]
```diff
@@ -412,3 +412,141 @@ trix-toolbar .button_group button.icon, trix-toolbar .button_group input[type=bu
...
+			}
+		}
+		.figure-row {
+			padding-bottom: 0.25rem;
+			margin-bottom: 0.5rem;
+			border-bottom: 1px solid rgba(0,0,0,0.1);
+		}
+	}
+
+	#table_report {
```
Comment (furilo) [inline app/assets/stylesheets/module-admin.scss]: This shouldn't go here. Look at comp-tables.scss - extend from there.

File: app/assets/stylesheets/module-admin.scss [hunk 4]
```diff
@@ -412,3 +412,141 @@ trix-toolbar .button_group button.icon, trix-toolbar .button_group input[type=bu
...
+			.tick line {
+				stroke-dasharray: 2, 2;
+				stroke: $text_soft;
+			}
+		}
+		.bars rect {
+			fill: #00909E;
+		}
+	}
+	#consultation-indicators {
```
Comment (furilo) [inline app/assets/stylesheets/module-admin.scss]: We are not reusing/extending existing styles from comp-indicator_widget?

File: app/assets/javascripts/admin/application.js (hunk 1 above)
Comment (martgnz) [reply app/assets/javascripts/admin/application.js]: I can't see that folder inside `javascripts/admin`. There's a `javascripts/gobierto_budget_consultations` folder, but it has the Vue app.

**Comment by martgnz** [comment]
I extended the components and reorganized the JS and now there are few lines of custom styles in module-admin. I think this is ready for another review.

**Comment by ferblape** [comment]
It's already integrated and ready to be reviewed in staging, i.e.: http://sencelles.gobify.net/admin/budgets/consultations/3/responses

(figures don't sum 100% because there are some wrong answers in staging)

I don't fully like the colors of the table, should we highlight the whole row on mouseover?

File: app/assets/stylesheets/comp-tables.scss [hunk 5]
```diff
@@ -156,6 +156,9 @@ table.block_content {
 		color: transparentize($color_main, .5);
 		border: none;
 	}
+  th.center {
```
Comment (furilo) [inline app/assets/stylesheets/comp-tables.scss:159]: why this if we already have a .center class in populities?

**Comment by furilo** [comment]
- [x] Print version is missing: Remove sidebar, make elements fit in the least possible space
- [x] Places don't fit in a laptop screen (see screenshot). Age distribution graphic would fit in less width; maybe we can create a 4 column grid, move the Place card to the second row, and have the age distribution in a 3 column width?
- [x] @ferblape: I would make the Report tab only appear once the consultation is finished, and it would be the first tab. 

captura de pantalla 2017-03-19 a las 11 06 33

**Comment by martgnz** [comment]
The print version and the 4 column grid are ready now, I think. Could you check again, please?

**Pull request merged**